docs/scaffold/band.md

```diff
@@ -0,0 +1,248 @@
+---
+order: 7
+description: IBC oracle integration with BandChain
+---
+
+# BandChain Oracle Scaffold
+
+BandChain’s Oracle module is a communication module built-in compliance with IBC protocol which can query data points of various types from BandChain. In addition, other chains can ask our Oracle module for real-time information according to their needs.
+BandChain has multiples scripts deployed into the network, and we can request any data using the script id.
+
+## IBC Module Packet Scaffold
+
+BandChain oracle queries can be scaffolded only in IBC modules.
+
+To scaffold an oracle:
+
+```
+starport scaffold band [queryName] --module [moduleName]
+```
+
+### Acknowledgement
+
+The BandChain oracle will return the ack messages with the request's id, and we save the last request id for future queries.
+
+## Files and Directories
+
+When you scaffold a BandChain oracle, the following files and directories are created and modified:
+
+- `proto`: oracle request and response data.
+- `x/module_name/keeper`: IBC hooks, gRPC message server.
+- `x/module_name/types`: message types, IBC events.
+- `x/module_name/client/cli`: CLI command to broadcast a transaction containing a message with a packet.
+- `x/module_name/oracle.go`: BandChain oracle packet handlers.
+
+## BandChain Oracle Scaffold Example
+
+The following command scaffolds the IBC-enabled oracle. by default, the starport scaffold oracle for [coin rates](https://laozi-testnet2.cosmoscan.io/oracle-script/37#bridge) request and result.
+
+```shell
+$ starport scaffold chain github.com/test/ibcoracle && cd ibcoracle 
+$ starport scaffold module consuming --ibc
+$ starport s band coinRates --module consuming
+```
+
+Note: BandChain module uses version "bandchain-1". Make sure to update the `keys.go` file accordingly.
+
+`x/ibcoracle/types/keys.go`
+```go
+const Version = "bandchain-1"
+```
+
+After scaffold and change the data, configure and run the starport relayer.
+```shell
+$ starport relayer configure -a \
+--source-rpc "http://rpc-laozi-testnet2.bandchain.org:26657" \
+--source-faucet "https://laozi-testnet2.bandchain.org/faucet/request" \
+--source-port "oracle" \
+--source-gasprice "0uband" \
+--source-gaslimit 5000000 \
+--source-prefix "band" \
+--source-version "bandchain-1" \
+--target-rpc "http://localhost:26657" \
+--target-faucet "http://localhost:4500" \
+--target-port "consuming" \
+--target-gasprice "0.0stake" \
+--target-gaslimit 300000 \
```

Comment:
Is this value sufficient for the target?

I did all my tests with `3000000`

Reply:
I think so!! we need more gas only on the Bandchain side. The @fadeev made this config